src/System.Linq.Parallel/src/System/Linq/Parallel/Utils/PairComparer.cs

```diff
@@ -37,6 +37,9 @@ public int Compare(Pair<T, U> x, Pair<T, U> y)
                 return result1;
             }
 
+            if (_comparer2 == null)
+                return result1;
```

Comment:
There is a case where comparer2 init-ed with null, thought better to check null, maybe just Assert?